blottertrax/helper/title_parser.py

```diff
@@ -1,53 +1,115 @@
-import re
-
-from blottertrax.value_objects.parsed_submission import ParsedSubmission
-
-
-class TitleParser:
-
-    @staticmethod
-    def create_parsed_submission_from_submission(submission) -> ParsedSubmission:
-        artist = None
-        feature_artist = None
-        song_title = None
-        post_title = submission.title
-
-        for dash in [' - ', ' — ', ' -- ', ' - - ', ' —— ']:
-            if dash in post_title:
-                split_result = post_title.split(dash)
-                artist = split_result[0]
-                song_title = split_result[1]
-
-                break
-
-        for feature in [' feat. ', ' ft. ', ' feature ', ' featuring ', ' & ']:
-            if feature not in post_title:
-                continue
-
-            feat_index = post_title.index(feature)
-
-            # remove featuring from artist if exists
-            if len(artist) > feat_index:
-                artist = artist[:feat_index].strip()
-
-            feat_index += len(feature)
-
-            # isolate featuring artist
-            feature_artist = post_title[feat_index:].strip()
-
-            for end_char in [' -', ')', '[', ' —']:
-                if end_char in feature_artist:
-                    split_result = feature_artist.split(end_char)
-                    feature_artist, song_title = split_result
-
-                    break
-
-        # Remove everything between () and []
-        song_title = re.sub(r"\(.*\)", "", song_title)
-        song_title = re.sub(r"[.*]", "", song_title)
-        song_title = song_title.strip()
-
-        if artist is not None:
-            return ParsedSubmission(artist, feature_artist, song_title, submission.url)
-
-        raise LookupError
+from blottertrax.value_objects.parsed_submission import ParsedSubmission
```

Comment:
This was written before the current one, and I did not go back to the current one because this one is more extensive anyway.

Reply:
I'm having a hard time understanding the title parser, this was also a reason why I rewrote parts of it so I could understand it myself. I don't think this is the way to go for this. We can improve on it later but it would be hard to maintain imo

Reply:
That one quite frankly is far away from working properly. I will make a new PR for the title parser changes once i have gone through and thoroughly commented. It's mostly quite simple code.